optuna/study/study.py

```diff
@@ -971,6 +977,48 @@ def objective(trial):
         for trial in trials:
             self.add_trial(trial)
 
+    def set_metric_names(self, metric_names: List[str]) -> None:
```

Comment:
Should we provide a property method like below?

```python
class Study:
    @property
    def metric_names() -> list[str]:
        ...
```

Reply:
I'm OK to introduce this type of new property to the `Study`, but it is arguable since the `Study` is the core API of Optuna. Let me discuss this after some users want to this feature.

Reply:
Sure. Let me resolve this.